### DISCUSSION TITLE
Installing vs2022 fresh with Maui Templates

### DISCUSSION THREAD

**Discussion by MeestorX** [dotnet/maui#1791] [General]
At first, maui-check didn't work at all, as it just closes if it doesn't find the .NET 5.0 runtime. Shouldn't it use .NET 6.0 so I don't have to have that installed?

Anyway, once I installed .NET 5.0 runtime, it checked everything, and I get green for all, but it still errors out at the end with "There were one or more problems detected..." Except it doesn't say what the problem(s) are!

https://user-images.githubusercontent.com/29129658/126707579-04efd438-e1dc-4c72-b85e-7d7fdc77ebc5.png

I see no Maui templates in VS2022 so something's still missing...

**Comment by smolinacadena** [comment]
Hi @MeestorX 

I believe that the templates are not yet supported in this preview for VS2022 but they will in the future!

For now, maui-check should help you install those so you can use it directly from the dotnet CLI. If for some reasons, they don't get installed, you can use the following:

`dotnet new --install Microsoft.Maui.Templates`

And once they're installed, you can run something like this:

`dotnet new maui -n HelloPreview6`

**Reply by MeestorX** [reply]
Thank you for your reply.
I don't understand why maui-check did not complete, but I opened a ticket on his GitHub site so perhaps he can answer that one.
I'll try the manual install as you suggest. Thx!

**Comment by MeestorX** [comment]
Still not seeing any Maui Templates, but dotnet new creates an app. 
Can MacOS apps be built using VS2022?

**Comment by BreyerW** [comment]
Actually templates show up after proper install did you use dotnet worklad install maui or something like that? (Exact command should be in guide) i did it from empty vs 2022 (no code option) and power shell command with admin privileges (run as admin on vs 2022)

However you have to install single project msix extension too but i encounter 500 error when downloading it ;/

**Reply by MeestorX** [reply]
Thank you for your reply, @BreyerW !
There's no selection for "maui" workload in the VS2022 installer.
What "guide" are you speaking of?

**Reply by BreyerW** [reply]
this https://devblogs.microsoft.com/dotnet/announcing-net-maui-preview-6/

after this you should be able to select maui template when firing vs 2022 (i can)

however you wont be able to play with maui anyway since you have to install single project msix package extension which is currently noninstallable due to 500 error

**Reply by MeestorX** [reply]
Aha! I thought you were talking about workloads in the installer.
Thank you for that link.
I didn't have any trouble with the msix package install.

**Reply by BreyerW** [reply]
Ok did that guide fixed issue for u ? And did you install extension before friday CET? Thats when i tried and still getting 500 error

**Comment by MeestorX** [comment]
`dotnet workload install maui ` fails part way through and then rolls back.
`maui-check` still errors out with no explanation.

What a mess.

**Reply by BreyerW** [reply]
for maui check u gotta go back and read all messages. i had error with android emulator not installed but in the middle of msg spam. Super annoying but thats how it currently works.

**Reply by MeestorX** [reply]
You can see the messages in my OP. What error am I missing?

**Reply by BreyerW** [reply]
this is suprising that there is no error msg but you probably can ignore it since everything seems green. As for workload install do you get any error msg?

EDIT unless u accidentally cropped out more messages from maui-check but i assume ealier in the console is only maui text-based logo?

**Comment by MeestorX** [comment]
Maui workload is not installed. Maui templates are not installed.
